templates/deployment.yml

```diff
@@ -29,11 +29,25 @@ spec:
       - image: armory/agent-k8s:{{ .Chart.AppVersion }}
         imagePullPolicy: IfNotPresent
         name: agent
-        {{- if .Values.cloudEnabled }}
         env:
```

Comment:
I tested when cloudEnabled and proxy settings are undefined, Kubernetes at least as of 1.18 doesn't care if this is left empty (e.g. no env variables are defined).